core/src/test/java/io/lionweb/lioncore/java/serialization/SerializationOfStructuredDataTypeInstancesTest.java

```diff
@@ -158,4 +161,54 @@ public void deserializeStructuredDataTypeInstance() {
     List<Node> deserialized = jsonSerialization.deserializeToNodes(serialized);
     assertEquals(Arrays.asList(n1), deserialized);
   }
+
+  @Test
+  public void serializeAndDeserializationMultiLevelStructuredDataTypeInstances() {
+    DynamicStructuredDataTypeInstance value =
+        new DynamicStructuredDataTypeInstance(MyNodeWithAmount.DECIMAL);
+    StructuredDataTypeInstanceUtils.setFieldValueByName(value, "int", 2);
+    StructuredDataTypeInstanceUtils.setFieldValueByName(value, "frac", 3);
+
+    DynamicStructuredDataTypeInstance amount =
+        new DynamicStructuredDataTypeInstance(MyNodeWithAmount.AMOUNT);
+    StructuredDataTypeInstanceUtils.setFieldValueByName(amount, "value", value);
+    StructuredDataTypeInstanceUtils.setFieldValueByName(
+        amount,
+        "currency",
+        new EnumerationValueImpl(MyNodeWithAmount.CURRENCY.getLiterals().get(0)));
+    StructuredDataTypeInstanceUtils.setFieldValueByName(amount, "digital", true);
+
+    MyNodeWithAmount n1 = new MyNodeWithAmount("n1");
+    n1.setAmount(amount);
+    assertEquals(amount, n1.getAmount());
+
+    JsonSerialization jsonSerialization = SerializationProvider.getStandardJsonSerialization();
+    jsonSerialization.registerLanguage(MyNodeWithAmount.LANGUAGE);
+    jsonSerialization.enableDynamicNodes();
+    String currencySerialized =
+        jsonSerialization
+            .getPrimitiveValuesSerialization()
+            .serialize(
+                MyNodeWithAmount.CURRENCY.getID(),
+                new EnumerationValueImpl(MyNodeWithAmount.CURRENCY.getLiterals().get(0)));
+    assertEquals("euro", currencySerialized);
+    String amountSerialized =
+        jsonSerialization
+            .getPrimitiveValuesSerialization()
+            .serialize(MyNodeWithAmount.AMOUNT.getID(), amount);
+    assertEquals(
+        "{\"value\":\"{\\\"int\\\":\\\"2\\\",\\\"frac\\\":\\\"3\\\"}\",\"currency\":\"euro\",\"digital\":\"true\"}",
```

Comment:
Is this according to spec? Decoding the escapes is hard without tooling.
Spec says to encode the whole json tree once, not each sub-tree: https://github.com/LionWeb-io/specification/blob/main/serialization/serialization.adoc#structured-datatype